### ISSUE TITLE
Broken attached property detection with `AddOwner` usage

### ISSUE THREAD

**Issue by MarchingCube** [AvaloniaUI/Avalonia#7925]
**Describe the bug**
After recent text processing changes `TextBlock` attached properties have been moved to `TextElement`. One would expect that old code using `TextBlock` properties will fail to compile so it can be fixed. But this is not the case and many invalid usages work fine.

**To Reproduce**

```xaml
<Window.Styles>
  <Style Selector="ContentPresenter">
    <Setter Property="TextBlock.FontWeight" Value="Normal" />
  </Style>
</Window.Styles>
  
<StackPanel>
  <ContentPresenter TextBlock.FontWeight="{DynamicResource ComboBoxHeaderThemeFontWeight}" />
  <ContentPresenter TextBlock.FontWeight="Normal" />
</StackPanel>
```

Usage in style setter - works fine
Usage with a dynamic resource or binding - works fine
Usage with a raw value - compilation error

**Expected behavior**
I would expect all cases above to error out since this is no longer an attached property (not sure about selector though) on `TextBlock`.

After a quick check in the compiler it seems that this property is detected as an attached property since property type name does not contain `Attached`. https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlTransformInstanceAttachedProperties.cs#L45

`AddOwner` on an attached property is supposed to convert it to a regular styled property I think but since property analysis is static it is not picked up. But even in runtime one cannot tell if this a converted attached property since original attached property is returned. https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/AttachedProperty.cs#L39

I think that user code will struggle a lot with random compilation errors caused by text breaking change until this issue gets resolved.

**Comment by robloo** [comment]
I can't compile master on my end due to what I assume are similar issues. Compiled bindings are throwing exceptions in HamburgerMenu for the Foreground property.

**Comment by robloo** [comment]
> Could it be some problem with VS build cache? Can you [cleanup](https://github.com/sebnilsson/DotnetCleanup) solution and try again (I can't believe that .NET still doesn't have cleanup tool build in).

Yes, as discussed in IM, build cache was the issue (likely related with custom project build configurations and switching branches).

**Comment by robloo** [comment]
... So anyway, reading through this issue again what you are saying is AddOwner is a bad idea in this case? @maxkatz6 this proves some of my earlier points :)

**Comment by MarchingCube** [comment]
It might not be a bad idea but it seems that currently `AddOwner` on an attached property causes weirdness.

` <ContentPresenter TextBlock.FontWeight="Normal" />`
![image](https://user-images.githubusercontent.com/2588062/161701715-18cc59ca-70cc-4b79-89c6-2c9407b543c8.png)

actually sets `TextElement.FontWeight` attached property in runtime even though according to the comment on `AddOwner` it is not supposed to be attached.

`<TextBlock FontWeight="Normal" />` sets ` TextElement.FontWeight` attached property too.

About my examples above:
- Access in a style setter works and sets an attached property. Technically we have a special syntax for that `(TextBlock.FontWeight)` but either one works in this case.
- Access via setters on control (`TextBlock.FontWeight="Normal"`) does not work even though technically it should (since `TextBlock.FontWeight` is just an alias for `TextElement.FontWeight`)
- Access via setters on control that use bindings ( (`TextBlock.FontWeight="{Binding}"`)) understand that this is an alias and work.

**Comment by MarchingCube** [comment]
My worry is coming from that my project has tons of code like:
![image](https://user-images.githubusercontent.com/2588062/161705771-01adbf58-6663-47c9-bafb-4e2d753f2831.png)

This seems to work for now. Now somebody changes one value to a non-binding.

![image](https://user-images.githubusercontent.com/2588062/161706083-05e979d5-de20-4db9-8c6a-4b533528095e.png)

And you get an interesting error message:

![image](https://user-images.githubusercontent.com/2588062/161706321-fb8695e8-20e8-44b7-9d52-afd681b839c5.png)

Which claims that one cannot set `FontSize` on a `TextBlock`.

**Comment by hez2010** [comment]
> Which claims that one cannot set FontSize on a TextBlock.

Hit the same issue with master branch.

**Comment by Gillibald** [comment]
TextBlock.FontSize etc. are not meant to be attached ony TextElement.FontSize should be an attached property. So you need to rewrite those usages like I did in the Fluent and Default theme.

**Comment by MarchingCube** [comment]
Yeah, I'm updating usages but it seems all tooling and compiler are a bit confused about current setup. Old attached properties shouldn't work anymore but everything happily continues. My problem is not with your change but with compiler/tooling not understanding this "borrowed" attached property and treating it as an attached property still.

**Comment by Gillibald** [comment]
Compiler also relies on static helper to get and set values for attached properties: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Documents/TextElement.cs#L209 these are required

**Comment by grokys** [comment]
Spoken privately with @MarchingCube and this problem was caused by referencing a stale assembly. Keeping the issue open though because in investigating the problem I found a slight discrepancy on how attached properties are resolved. PR incoming.
